core/embed/trezorhal/unix/touch.c

```diff
@@ -53,227 +51,230 @@ typedef enum {
         }
         break;
       case SDLK_RIGHT:
-        if (input_state == BUTTON_SWIPE_RIGHT_INITIATED) {
+        if (driver->state == BUTTON_SWIPE_RIGHT_INITIATED) {
           *ev_x = sdl_display_res_x - _btn_swipe_begin - _btn_swipe_length;
           *ev_y = sdl_display_res_y / 2;
           *ev_type = TOUCH_MOVE;
-          input_state = BUTTON_SWIPE_COMPLETED;
+          driver->state = BUTTON_SWIPE_COMPLETED;
         }
         break;
       case SDLK_UP:
-        if (input_state == BUTTON_SWIPE_UP_INITIATED) {
+        if (driver->state == BUTTON_SWIPE_UP_INITIATED) {
           *ev_x = sdl_display_res_x / 2;
           *ev_y = _btn_swipe_begin + _btn_swipe_length;
           *ev_type = TOUCH_MOVE;
-          input_state = BUTTON_SWIPE_COMPLETED;
+          driver->state = BUTTON_SWIPE_COMPLETED;
         }
         break;
       case SDLK_DOWN:
-        if (input_state == BUTTON_SWIPE_DOWN_INITIATED) {
+        if (driver->state == BUTTON_SWIPE_DOWN_INITIATED) {
           *ev_x = sdl_display_res_x / 2;
           *ev_y = sdl_display_res_y - _btn_swipe_begin - _btn_swipe_length;
           *ev_type = TOUCH_MOVE;
-          input_state = BUTTON_SWIPE_COMPLETED;
+          driver->state = BUTTON_SWIPE_COMPLETED;
         }
         break;
     }
   }
 }
 
-uint32_t touch_read(void) {
-  if (input_state == BUTTON_SWIPE_COMPLETED) {
-    input_state = IDLE;
-    return TOUCH_END | touch_pack_xy(_touch_x, _touch_y);
-  }
+secbool touch_init(void) {
+  touch_driver_t* driver = &g_touch_driver;
 
-  emulator_poll_events();
-  SDL_Event event;
-  SDL_PumpEvents();
+  if (driver->initialized != sectrue) {
+    memset(driver, 0, sizeof(touch_driver_t));
+    driver->state = IDLE;
+    driver->initialized = sectrue;
+  }
 
-  int ev_x = 0;
-  int ev_y = 0;
-  int ev_type = 0;
+  return driver->initialized;
+}
 
-  while (SDL_PollEvent(&event) > 0) {
-    if (input_state == IDLE || input_state == MOUSE_DOWN_INSIDE ||
-        input_state == MOUSE_DOWN_OUTSIDE) {
-      handle_mouse_events(event, &ev_type, &ev_x, &ev_y);
-    }
-    if (input_state == IDLE || is_button_swipe_initiated()) {
-      handle_button_events(event, &ev_type, &ev_x, &ev_y);
-    }
+void touch_deinit(void) {
+  touch_driver_t* driver = &g_touch_driver;
 
-    if (ev_type != 0) {
-      _touch_x = ev_x;
-      _touch_y = ev_y;
-    }
+  if (driver->initialized == sectrue) {
+    memset(driver, 0, sizeof(touch_driver_t));
   }
-  return ev_type | touch_pack_xy(ev_x, ev_y);
 }
 
-secbool touch_init(void) { return sectrue; }
-void touch_power_on(void) {}
-void touch_wait_until_ready(void) {}
-
-uint32_t touch_is_detected(void) {
-  return input_state == MOUSE_DOWN_INSIDE || is_button_swipe_initiated();
+secbool touch_ready(void) {
+  touch_driver_t* driver = &g_touch_driver;
+  return driver->initialized;
 }
 
-uint8_t touch_get_version(void) { return 0; }
-
-#endif
+secbool touch_set_sensitivity(uint8_t value) {
+  // Not implemented on the simulator
```

Comment:
lets keep calling it emulator in code/comments

Reply:
fixed, see c04b50822941a7c6f979e8923827122438fe5749